trollsched/shapefiles_from_schedule.py

```diff
@@ -0,0 +1,157 @@
+from pathlib import Path
+
+import defusedxml.ElementTree as ET
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import pyproj
+from shapely.geometry import Polygon
+from shapely.ops import transform
+
+from trollsched import INSTRUMENT, SATELLITE_NAMES
+from trollsched.logger import setup_logging
+from trollsched.satpass import create_pass
+
+logger = logging.getLogger(__name__)
+
+
+def get_shapely_polygon_from_lonlat(lons, lats):
+    """From a arrays of lons and lats return a shapely Polygon."""
+    geodata = np.vstack((lons, lats)).T
+    return Polygon(np.rad2deg(geodata))
+
+
+def create_shapefile_filename(satellite_pass_obj):
+    """From a trollsched.satpass instance create the shapefile filename."""
+    satname = satellite_pass_obj.satellite.name.replace(" ", "-")
+    prefix = f"{satellite_pass_obj.instrument}_{satname}"
+    return f"{prefix}_{satellite_pass_obj.risetime:%Y%m%d%H%M}_{satellite_pass_obj.falltime:%Y%m%d%H%M}_outline.shp"
+
+
+def create_shapefile_from_pass(sat_pass, output_filepath):
+    """From a satellite overpass (instrument scanning outline) create a shapefile and save."""
+    sat_poly = get_shapely_polygon_from_lonlat(sat_pass.boundary.contour_poly.lon,
+                                               sat_pass.boundary.contour_poly.lat)
+
+    wgs84 = pyproj.CRS("EPSG:4326")  # WGS 84
+    a_crs = {"init": "epsg:4326"}
+
+    project = pyproj.Transformer.from_crs(wgs84, a_crs, always_xy=True).transform
+    new_shapes = transform(project, sat_poly)
+
+    gpd.GeoDataFrame(pd.DataFrame(["p1"], columns=["geom"]), crs=a_crs,
+                     geometry=[new_shapes]).to_file(output_filepath)
+
+
+def shapefiles_from_schedule_xml_requests(filename, satellites, tle_file, output_dir):
+    """From XML file with planned satellite passes create shapefiles for each Pass outline.
+
+    Open the XML file with the scheduled satellite pass plan, and create
+    shapefiles for each Pass outline assuming the instrument as provided in the
+    INSTRUMENTS dictionary.
+    """
+    tree = ET.parse(filename)
+    root = tree.getroot()
+
+    for child in root:
+        if child.tag == "pass":
+            platform_name = SATELLITE_NAMES.get(child.attrib["satellite"], child.attrib["satellite"])
+            instrument = INSTRUMENT.get(platform_name)
+            if not instrument:
+                continue
+
+            if platform_name not in satellites:
+                continue
+            try:
+                overpass = create_pass(platform_name, instrument,
+                                       datetime.strptime(child.attrib["start-time"],
+                                                         "%Y-%m-%d-%H:%M:%S"),
+                                       datetime.strptime(child.attrib["end-time"],
+                                                         "%Y-%m-%d-%H:%M:%S"),
+                                       tle_filename=tle_file)
+            except KeyError:
+                continue
```

Comment:
imo these `continue`s should be replaced by log messages.

Reply:
Not all of them!? Not "replaced"? I don't want to try create a pass if the instrument is None or the platform-name not requested...
How do you mean?

Reply:
I mean that if one of these continues is reached, something is not going as planned, right? so shouldn’t we log about it?

Reply:
I believe this is fixed now

Reply:
I think we need to figure out if the the KeyError part can be tested and when/why that actually might happen?